libs/llrb/llrb.go

```diff
@@ -0,0 +1,266 @@
+package llrb
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+)
+
+// MaxSize is the max allowed number of node a llrb is allowed to contain
+const MaxSize = int(^uint(0) >> 1)
+
+func (a NodeKey) compare(b NodeKey) int {
+	if a.Priority > b.Priority {
+		return 1
+	}
+	if a.Priority < b.Priority {
+		return -1
+	}
+	if a.TS.Before(b.TS) {
+		return 1
+	}
+	if a.TS.After(b.TS) {
+		return -1
+	}
+	return 0
+}
+
+type node struct {
+	key         NodeKey
+	data        interface{}
+	left, right *node
+	black       bool
+}
+
+type llrb struct {
+	mtx     sync.RWMutex
+	wg      *sync.WaitGroup
```

Comment:
seems we are not using this?

Reply:
Do we need to use wg in insert or remove for the concurrent queries?